### ISSUE TITLE
Getting error while setting "Value" attribute in hidden input form using <%= or <%- tag

### ISSUE THREAD

**Issue by gabo71096** [mde/ejs#637]
I'm building an ecommerce web app attempt, and never had any issues with the <%= tag, I have srcs, hrefs and many attributes assigned with EJS, but when I try to use it in the value attribute in a form, I get an error telling me that "id is not defined" (id is the name that I use when rendering the form from NodeJS using Express).
The weird thing is that after getting the 500 error (logged using Morgan), I still get to add a document to my MongoDB database using the data from the form, including the hidden "id" that's getting me the error.
Also I can see the hidden value in the form page using developer tools, it's included correctly.

![devTools](https://user-images.githubusercontent.com/18217269/139558315-f58a19d9-b311-4769-9f8f-ef80653653af.png)
![errorPage](https://user-images.githubusercontent.com/18217269/139558327-f7c856fa-6eed-4890-abf0-fe763cd1804a.png)
![error500_dbInsertion](https://user-images.githubusercontent.com/18217269/139558330-584271c1-e1ca-44e0-a051-53dba075e58e.png)
![ejsFormPage](https://user-images.githubusercontent.com/18217269/139558332-b72e1fce-e9fa-4007-aa06-831f7ac8c688.png)

**Comment by mde** [comment]
EJS really just interprets JavaScript. The only reason I can think for getting that error is that the `id` variable in your locals is genuinely `undefined`. How are you invoking the EJS `render`, and have you verified that that variable is defined in your locals?

**Comment by gabo71096** [comment]
First of all, thanks for the response and for your hard work!
I checked many times the POST route, because that's the one that was giving me the 500 error, and realize that after the insertion, I rendered again the same page, but I didn't include the id in that sentence. Hence, the undefined error.

I'm sorry if I took time from you, and I thank you greatly again for your help.

![image](https://user-images.githubusercontent.com/18217269/139727216-1d632442-4fc3-428b-8bb8-a32525cc87be.png)
